File: spring-boot-3tier-form/src/main/java/com/adjh/springboot3tierform/service/UserService.java
package com.adjh.springboot3tierform.service;

import com.adjh.springboot3tierform.model.dto.UserDto;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Please explain the class!!
 *
 * @author : jonghoon
 * @fileName : UserService
 * @since : 5/15/24
 */
@Service
public interface UserService {

    List<UserDto> selectUserList(UserDto userDto);

    int insertUser(UserDto userDto);

    int updateUser(UserDto userDto);

    int deleteUser(UserDto userDto);
}

File: kotlin-spring-boot-form/src/main/kotlin/com/blog/kotlinspringbootform/service/UserService.kt
package com.blog.kotlinspringbootform.service

import com.blog.kotlinspringbootform.model.dto.UserDto
import org.springframework.stereotype.Service

@Service
interface UserService {

    // 추상 메서드
    fun selectUserList(userDto: UserDto): List<UserDto>

    // 추상 메서드
    fun insertUser(userDto: UserDto): Int

    // 추상 메서드
    fun updateUser(userDto: UserDto): Int

    // 추상 메서드
    fun deleteUserList(userDto: List<UserDto>)

    // 디폴트 메서드
    fun showDefaultMethod(userDto: UserDto) {
        println("default Method...")
    }
}

